### ISSUE TITLE
Can guidance support batch input?

### ISSUE THREAD

**Issue by Tangent-90C** [guidance-ai/guidance#819]
I've been using guidance to evaluate a dataset recently, but guidance can only handle one input at a time, which is really slow, and I was wondering if support for batch_input would improve guidance's throughput and speed up my evaluation of a dataset?

**Comment by Harsha-Nori** [comment]
Hey @Tangent-90C , not yet but we're starting to brainstorm about this. Can I ask if you're planning to use this on a local model (transformers/llamacpp) or remote or both? The implementation details change between the two and it'd be helpful to prioritize.

**Comment by Tangent-90C** [comment]
yes, and I prefer transformers because they support a lot of models. It would be better if it could support TurboMind. I always use TurboMind to run LLM evaluation, because its inference throughput is very large
